libs/http2-manager/test/Test/HTTP2/Client/ManagerSpec.hs

```diff
@@ -270,59 +268,51 @@ withTestServerOnSocket mCtx action (serverPort, listenSock) = do
     action TestServer {..}
 
-allocServerConfig :: Either Socket SSL.SSL -> IO Server.Config
-allocServerConfig (Left sock) = HTTP2.allocSimpleConfig sock 4096
-allocServerConfig (Right ssl) = do
-  buf <- mallocBytes bufsize
-  timmgr <- System.TimeManager.initialize $ 30 * 1000000
-  -- Sometimes the frame header says that the payload length is 0. Reading 0
-  -- bytes multiple times seems to be causing errors in openssl. I cannot figure
-  -- out why. The previous implementation didn't try to read from the socket
-  -- when trying to read 0 bytes, so special handling for 0 maintains that
-  -- behaviour.
-  let readData prevChunk 0 = pure prevChunk
-      readData prevChunk n = do
-        -- Handling SSL.ConnectionAbruptlyTerminated as a stream end
-        -- (some sites terminate SSL connection right after returning the data).
-        chunk <- SSL.read ssl n `catch` \(_ :: SSL.ConnectionAbruptlyTerminated) -> pure mempty
-        let chunkLen = BS.length chunk
-        if
-            | chunkLen == 0 || chunkLen == n ->
-                pure (prevChunk <> chunk)
-            | chunkLen > n ->
-                error "openssl: SSL.read returned more bytes than asked for, this is probably a bug"
-            | otherwise ->
-                readData (prevChunk <> chunk) (n - chunkLen)
-  pure
-    Server.Config
-      { Server.confWriteBuffer = buf,
-        Server.confBufferSize = bufsize,
-        Server.confSendAll = SSL.write ssl,
-        Server.confReadN = readData mempty,
-        Server.confPositionReadMaker = Server.defaultPositionReadMaker,
-        Server.confTimeoutManager = timmgr
+allocServerConfig :: (Socket, Maybe SSL.SSL) -> IO Server.Config
+allocServerConfig (sock, Nothing) =
+  HTTP2.allocSimpleConfig sock 4096
+allocServerConfig (sock, Just ssl) = do
+  config <- HTTP2.allocSimpleConfig sock 4096
+  pure $
+    config
+      { Server.confReadN = readData mempty,
+        Server.confSendAll = SSL.write ssl
       }
+  where
+    readData prevChunk 0 = pure prevChunk
+    readData prevChunk n = do
+      -- Handling SSL.ConnectionAbruptlyTerminated as a stream end
+      -- (some sites terminate SSL connection right after returning the data).
+      chunk <- SSL.read ssl n `catch` \(_ :: SSL.ConnectionAbruptlyTerminated) -> pure mempty
+      let chunkLen = BS.length chunk
+      if
+          | chunkLen == 0 || chunkLen == n ->
+              pure (prevChunk <> chunk)
+          | chunkLen > n ->
+              error "openssl: SSL.read returned more bytes than asked for, this is probably a bug"
+          | otherwise ->
+              readData (prevChunk <> chunk) (n - chunkLen)
 
 testServerOnSocket :: Maybe SSL.SSLContext -> Socket -> IORef Int -> IORef (Map Unique (Async ())) -> IO ()
 testServerOnSocket mCtx listenSock connsCounter conns = do
   listen listenSock 1024
   forever $ do
     (sock, _) <- accept listenSock
     serverCfgParam <- case mCtx of
```

Comment:
in that case you could just match here directly and then only pass the `snd` to the case